config/clients/java/template/OpenFgaApiTest.java.mustache

```diff
@@ -0,0 +1,709 @@
+        mockHttpClient
+                .verify()
+                .get("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X/authorization-models")
+                .called(1);
+        assertNotNull(response.getAuthorizationModels());
+        assertEquals(1, response.getAuthorizationModels().size());
+        AuthorizationModel authModel = response.getAuthorizationModels().get(0);
+        assertEquals(DEFAULT_AUTH_MODEL_ID, authModel.getId());
+        assertEquals(DEFAULT_SCHEMA_VERSION, authModel.getSchemaVersion());
+    }
+
+    @Test
+    public void readAuthorizationModels_storeIdRequired() throws ApiException {
+        // When
+        ApiException exception = assertThrows(ApiException.class, () -> fga.readAuthorizationModels(null, null, null));
+
+        // Then
+        assertEquals(
+                "Missing the required parameter 'storeId' when calling readAuthorizationModels",
+                exception.getMessage());
+    }
+
+    /**
+     * Return a list of all the tuple changes.
+     */
+    @Test
+    public void readChangesTest() throws ApiException {
+        // Given
+        mockHttpClient
+                .onGet("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X/changes")
+                .doReturn(
+                        200,
+                        String.format(
+                                "{\"changes\":[{\"tuple_key\":{\"object\":\"%s\",\"relation\":\"%s\",\"user\":\"%s\"}}]}",
+                                DEFAULT_OBJECT, DEFAULT_RELATION, DEFAULT_USER));
+
+        // When
+        ReadChangesResponse response = fga.readChanges(DEFAULT_STORE_ID, null, null, null);
+
+        // Then
+        mockHttpClient
+                .verify()
+                .get("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X/changes")
+                .called(1);
+        assertNotNull(response.getChanges());
+        assertEquals(1, response.getChanges().size());
+        TupleChange change = response.getChanges().get(0);
+        assertNotNull(change.getTupleKey());
+        assertEquals(DEFAULT_OBJECT, change.getTupleKey().getObject());
+        assertEquals(DEFAULT_RELATION, change.getTupleKey().getRelation());
+        assertEquals(DEFAULT_USER, change.getTupleKey().getUser());
+    }
+
+    @Test
+    public void readChanges_storeIdRequired() throws ApiException {
+        // When
+        ApiException exception = assertThrows(ApiException.class, () -> fga.readChanges(null, null, null, null));
+
+        // Then
+        assertEquals("Missing the required parameter 'storeId' when calling readChanges", exception.getMessage());
+    }
+
+    /**
+     * Add or delete tuples from the store.
+     */
+    @Test
+    public void writeTest() throws ApiException {
+        // Given
+        mockHttpClient
+                .onPost("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X/write")
+                .doReturn(200, "{}");
+
+        // When
+        fga.write(DEFAULT_STORE_ID, new WriteRequest());
```

Comment:
To make it more realistic, maybe a non-empty body esp. since this is a commonly used method?  Also perhaps 1 test for write and 1 test for delete?

Reply:
Ah I see, write can be both write or delete. I'll flesh this out to two tests

I can't provide more information in the response since there is no response. [The API contract for `write`](https://openfga.dev/api/service#/Relationship%20Tuples/Write) is that it should return 200 `{}` for everything except error responses, and the API is rendered (I think correctly) as having a `void` return type.



Actually, I don't have any error responses modeled in tests yet

Reply:
- [ ] When `"{}"` is the only valid response return it as a constant
- [ ] Flesh out `write` with both a write and delete request
- [ ] Add simple tests with error responses (Testing serialization)